lib/using_restricted_array.rb

```diff
@@ -6,56 +6,104 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(1), the block will always run at most 20 times.
```

Comment:
The block size depends on the size of the given string, it will **not** always run 20 times.  So this is O(n)